src/core/libraries/audio/sdl_audio.cpp

```diff
@@ -21,8 +22,41 @@ class SDLPortBackend : public PortBackend {
             .channels = port.format_info.num_channels,
             .freq = static_cast<int>(port.sample_rate),
         };
-        stream =
-            SDL_OpenAudioDeviceStream(SDL_AUDIO_DEVICE_DEFAULT_PLAYBACK, &fmt, nullptr, nullptr);
+
+        // Determine port type
+        std::string port_name = port.type == OrbisAudioOutPort::PadSpk
+                                    ? Config::getPadSpkOutputDevice()
+                                    : Config::getMainOutputDevice();
+        SDL_AudioDeviceID dev_id = 0;
+
+        if (port_name == "None") {
+            stream = nullptr;
+            return;
+        } else if (port_name == "Default Device") {
+            dev_id = SDL_AUDIO_DEVICE_DEFAULT_PLAYBACK;
+        } else {
+            try {
+                SDL_AudioDeviceID* dev_array = SDL_GetAudioPlaybackDevices(nullptr);
+                for (; dev_array != 0;) {
+                    std::string name = static_cast<std::string>(SDL_GetAudioDeviceName(*dev_array));
```

Comment:
```suggestion
                    std::string name(SDL_GetAudioDeviceName(*dev_array));
```

Should just construct from the result instead of casting a `const char*` to `std::string`.